File: typescript/src/patterns/behavioral/chainofresponsibility/Validator.ts
import { requireNonNull } from '../../../shared/requireNonNull.js';
import { Handler } from './Handler.js';

export class Validator extends Handler {
  constructor(nextHandler: Handler) {
    super();
    this.nextHandler = nextHandler;
  }

  handleRequest(request: string): void {
    if (request.length < 3) {
      return;
    }
    requireNonNull(this.nextHandler).handleRequest(request);
  }
}

File: csharp/src/Training.Patterns/Behavioral/ChainOfResponsibility/Validator.cs
namespace Training.Patterns.Behavioral.ChainOfResponsibility;

public class Validator : Handler
{
    public Validator(Handler nextHandler)
    {
        this.nextHandler = nextHandler;
    }

    public override void HandleRequest(string request)
    {
        if (request.Length < 3)
        {
            return;
        }
        nextHandler!.HandleRequest(request);
    }
}
